gitlab/__init__.py

```diff
@@ -342,8 +347,14 @@ def list(self, obj_class, **kwargs):
                 if key in cls_kwargs:
                     del cls_kwargs[key]
 
-            return [cls(self, item, **cls_kwargs) for item in r.json()
-                    if item is not None]
+            results = [cls(self, item, **cls_kwargs) for item in r.json()
+                       if item is not None]
+            if 'next' in r.links and 'url' in r.links['next']:
+                args = kwargs.copy()
+                args['next_url'] = r.links['next']['url']
+                logger.debug("Iterating results 'next' link: %s", args['next_url'])
+                results.extend(self.list(obj_class, **args))
```

Comment:
This doesn't seem to give a choice to the user, this will always return the full list and do multiple API calls. If a user has hundreds of items this is a problem. It should be optional.